public/javascripts/application.js

```diff
@@ -1,19 +1,52 @@
-$(document).on('click', '#secret_submit', function () {
-  $.post('/encrypt/get_ct_json', {
-    plaintext_secret: $('#plaintext_secret').val()
-  },
-  function(json) {
-    if (json.error) {
-      $('#alert').addClass('alert-error');
-      $('#alert').attr('style', 'display: block !important');
-      if (json.error.message) {
-        $('#alert').html(json.error.message);
+'use strict';
+
+document.addEventListener('DOMContentLoaded', function() {
+  var submit = document.querySelector('#secret_submit');
+  var alert = document.querySelector('#alert');
+  var response = document.querySelector('#response');
+
+  submit.addEventListener('click', function() {
+    var request = new XMLHttpRequest();
+    var data = {
+      plaintext_secret: document.querySelector('#plaintext_secret').value
+    };
+
+    request.open('POST', '/encrypt/get_ct_json', true);
+    request.setRequestHeader('Content-Type', 'application/json; charset=UTF-8');
+    request.onload = function() {
+      var resp = '';
+
+      try {
+        resp = JSON.parse(request.responseText);
+      } catch (ex) {}
+
+      if (request.status >= 200 && request.status < 400) {
+        alert.classList.add('hidden');
+        alert.innerHTML = '';
+
+        response.classList.remove('hidden');
+        response.innerHTML = '$tokend\n' +
+          '  type: transit\n' +
+          '  resource: /v1/transit/default/decrypt\n' +
+          '  key: ' + resp.key + '\n' +
+          '  ciphertext: "' + resp.ciphertext + '"\n';
       } else {
-        $('#alert').html('Check logs for stacktrace, error is undefined');
+        // Error case
+        response.classList.add('hidden');
+        response.innerHTML = '';
+
+        alert.classList.remove('hidden');
+        if (!resp.error) {
+          alert.innerHTML = 'Check logs for stacktrace, error is undefined';
+
+          return;
+        }
+
+        resp.error.errors.forEach(function(err) {
+          alert.innerHTML += err + '\n';
+        });
```

Comment:
ideally this could done just as a reduce

```javascript
alert.innerHTML = resp.error.errors.reduce(function (errorString, error) {
  return errorString + error + '\n';
}, '');
```

It took all of my will not to make this ES2015.